pallets/tx_fee/src/lib.rs

```diff
@@ -0,0 +1,57 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use frame_support::weights::{DispatchInfo, GetDispatchInfo};
+use pallet_transaction_payment::{FeeDetails, OnChargeTransaction};
+use sp_runtime::{traits::Dispatchable, FixedPointOperand};
+
+#[cfg(test)]
+mod tests;
+
+pub use pallet::*;
+
+// Type aliases used for interaction with `OnChargeTransaction`.
+pub(crate) type OnChargeTransactionOf<T> =
+	<T as pallet_transaction_payment::Config>::OnChargeTransaction;
+
+// Balance type alias.
+pub(crate) type BalanceOf<T> = <OnChargeTransactionOf<T> as OnChargeTransaction<T>>::Balance;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+
+	#[pallet::config]
+	pub trait Config: frame_system::Config + pallet_transaction_payment::Config {}
+
+	#[pallet::pallet]
+	#[pallet::generate_store(pub(super) trait Store)]
+	pub struct Pallet<T>(_);
+
+	#[pallet::error]
+	pub enum Error<T> {}
+
+	impl<T: Config> Pallet<T> {
+		pub fn compute_extrinsic_cost<Extrinsic: sp_runtime::traits::Extrinsic + GetDispatchInfo>(
```

Comment:
is there a method that computes cost already?

Reply:
Because we will have to add adjustments to basic cost but that is future work
And here I am using pallet transaction to calculate base cost , check the pallet function

Reply:
If this is future work lets merge it when we have that problem. We don't have that problem. I looked through other substrate base code and I could not find one that does this.

Like I mention I think we should solve for problems with have. At the moment we dont have this problem. Lets not create one.